modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -188,17 +195,60 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
    */
   private void shutdownMySqlAbandonedConnectionCleanupThread() {
 
-    Class<?> classAbandonedConnectionCleanupThread;
+    Class<?> classAbandonedConnectionCleanupThread = null;
     try {
-      classAbandonedConnectionCleanupThread =
-          this.getClass().getClassLoader().loadClass("com.mysql.jdbc.AbandonedConnectionCleanupThread");
+      for (String knownCleanupThreadClassAddress : CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES) {
+        try {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Extract this nested try block into a separate method. ([squid:S1141](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1141))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7685&issues=AWjzLS16otIXnmsUYkAr&open=AWjzLS16otIXnmsUYkAr)

[ISSUE_KEY]: AWjzLS16otIXnmsUYkAr